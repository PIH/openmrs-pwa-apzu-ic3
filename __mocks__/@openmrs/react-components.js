import types from 'redux-types';
// mock methods

const visitRest = {

  createVisit: (params) => {

    const visit = params.visit;

    if (visit.patient != 'patient_uuid' || visit.visitType != 'visit_type_uuid' || visit.location != 'location_uuid') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const encounterRest = {

  createEncounter: (params) => {

    if (params.encounter.encounterType === 'invalid_encounter_type') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const BASIC_TYPES = [
  'FETCH_SUCCEEDED',
  'FETCH_PENDING',
  'SET_SUCCEEDED',
  'SET_REQUESTED',
  'SET_FAILED',
  'FETCH_FAILED',
  'FETCH_REQUESTED',
  'SUCCEEDED',
  'PENDING',
  'FAILED',
  'REQUESTED'
];

const LOGIN_TYPES = {
  LOGIN: types('login', BASIC_TYPES),
  LOGIN_LOCATIONS: types('login_locations', BASIC_TYPES),
  LOGOUT: types('logout', BASIC_TYPES)
}

const SESSION_TYPES = types('session', BASIC_TYPES)


export {
  visitRest,
  encounterRest,
  LOGIN_TYPES,
  SESSION_TYPES
};
