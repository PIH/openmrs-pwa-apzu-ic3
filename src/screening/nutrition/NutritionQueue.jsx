import React from "react";
import { connect } from "react-redux";
import { push } from 'connected-react-router';
import { selectors } from '@openmrs/react-components';
import nutritionFilters from './nutritionFilters';
import ScreeningQueue from "../ScreeningQueue";

// TODO can we figure out a better way to do this without passing dispatch all the way through?

let NutritionQueue = props => {

  const rowSelectedActionCreators = [
    () => push('/screening'),
  ];

  return (
    <div>
      <ScreeningQueue
        dispatch={props.dispatch}
        filters={[nutritionFilters.required, (patient) => !nutritionFilters.completed(patient)]}
        rowData={Object.values(props.patients)}
        rowSelectedActionCreators={rowSelectedActionCreators}
        title="Nutrition Queue"
      />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    patients: selectors.getPatientStore(state),
  };
};

export default connect(mapStateToProps)(NutritionQueue);
