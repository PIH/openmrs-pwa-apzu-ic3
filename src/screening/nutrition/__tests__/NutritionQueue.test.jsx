import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import {DataGrid, visitActions, patientActions} from '@openmrs/react-components';
import NutritionQueue from '../NutritionQueue';
import {ACTIVE_VISITS_REP, VISIT_REPRESENTATION} from "../../../constants";

let props, store;
let mountedComponent;

const mockStore = configureMockStore();

const nutritionQueue = () => {
  if (!mountedComponent) {
    mountedComponent = mount(
      <Provider store={store}>
        <NutritionQueue {...props} />
      </Provider>);
  }
  return mountedComponent;
};

describe('Component: NutritionQueue', () => {
  beforeEach(() => {
    props = {
      session: {
        sessionLocation: {
          uuid: 'abc'
        }
      }};
    store = mockStore(
      {
        dispatch: {},
        openmrs: {
          session: {
            sessionLocation: {
              uuid: 'abc'
            }
          },
          patients: {
            set: {},
            isUpdating: false,
            selected: null
          }
        },
      });
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    expect(toJson(nutritionQueue())).toMatchSnapshot();
    expect(nutritionQueue().find(DataGrid).length).toBe(1);
    expect(nutritionQueue().find(DataGrid).props().rowSelectedActionCreators.length).toBe(2);
    expect(nutritionQueue().find(DataGrid).props().rowSelectedActionCreators[1]().payload.args[0]).toBe("/screening/nutrition/form");
    expect(store.getActions()).toContainEqual(patientActions.clearSelectedPatient());
    expect(store.getActions()).toContainEqual(visitActions.fetchActiveVisits(props.session.sessionLocation.uuid, ACTIVE_VISITS_REP));
  });

});
