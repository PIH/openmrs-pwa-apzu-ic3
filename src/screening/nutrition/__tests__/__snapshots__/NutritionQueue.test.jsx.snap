// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NutritionQueue renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(NutritionQueue)
    session={
      Object {
        "sessionLocation": Object {
          "uuid": "abc",
        },
      }
    }
  >
    <NutritionQueue
      dispatch={[Function]}
      patients={Array []}
      session={
        Object {
          "sessionLocation": Object {
            "uuid": "abc",
          },
        }
      }
    >
      <div>
        <Connect(ScreeningQueue)
          dispatch={[Function]}
          filters={
            Array [
              [Function],
            ]
          }
          rowData={Array []}
          rowSelectedActionCreators={
            Array [
              [Function],
            ]
          }
          title="Nutrition Queue"
        >
          <ScreeningQueue
            columnDefs={
              Array [
                Object {
                  "field": "uuid",
                  "headerName": "uuid",
                  "hide": true,
                },
                Object {
                  "autoHeight": true,
                  "cellRenderer": [Function],
                  "headerName": "Id",
                },
                Object {
                  "field": "name.givenName",
                  "headerName": "Given Name",
                },
                Object {
                  "field": "name.familyName",
                  "headerName": "Family Name",
                },
                Object {
                  "field": "gender",
                  "headerName": "Gender",
                },
                Object {
                  "field": "age",
                  "headerName": "Age",
                },
                Object {
                  "headerName": "Checked-in Time",
                  "valueGetter": [Function],
                },
              ]
            }
            dispatch={[Function]}
            filters={
              Array [
                [Function],
              ]
            }
            rowData={Array []}
            rowSelectedActionCreators={
              Array [
                [Function],
              ]
            }
            session={
              Object {
                "sessionLocation": Object {
                  "uuid": "abc",
                },
              }
            }
            title="Nutrition Queue"
          >
            <div>
              <List
                columnDefs={
                  Array [
                    Object {
                      "field": "uuid",
                      "headerName": "uuid",
                      "hide": true,
                    },
                    Object {
                      "autoHeight": true,
                      "cellRenderer": [Function],
                      "headerName": "Id",
                    },
                    Object {
                      "field": "name.givenName",
                      "headerName": "Given Name",
                    },
                    Object {
                      "field": "name.familyName",
                      "headerName": "Family Name",
                    },
                    Object {
                      "field": "gender",
                      "headerName": "Gender",
                    },
                    Object {
                      "field": "age",
                      "headerName": "Age",
                    },
                    Object {
                      "headerName": "Checked-in Time",
                      "valueGetter": [Function],
                    },
                  ]
                }
                delayInterval={10000}
                fetchListActionCreator={[Function]}
                filters={
                  Array [
                    [Function],
                    [Function],
                  ]
                }
                onMountOtherActionCreators={
                  Array [
                    [Function],
                  ]
                }
                rowData={Array []}
                rowSelectedActionCreators={
                  Array [
                    [Function],
                  ]
                }
                title="Nutrition Queue"
              >
                <div>
                  <h3>
                    <Label
                      bsClass="label"
                      bsStyle="default"
                    >
                      <span
                        className="label label-default"
                      >
                        Nutrition Queue
                      </span>
                    </Label>
                  </h3>
                  <Connect(DataGrid)
                    columnDefs={
                      Array [
                        Object {
                          "field": "uuid",
                          "headerName": "uuid",
                          "hide": true,
                        },
                        Object {
                          "autoHeight": true,
                          "cellRenderer": [Function],
                          "headerName": "Id",
                        },
                        Object {
                          "field": "name.givenName",
                          "headerName": "Given Name",
                        },
                        Object {
                          "field": "name.familyName",
                          "headerName": "Family Name",
                        },
                        Object {
                          "field": "gender",
                          "headerName": "Gender",
                        },
                        Object {
                          "field": "age",
                          "headerName": "Age",
                        },
                        Object {
                          "headerName": "Checked-in Time",
                          "valueGetter": [Function],
                        },
                      ]
                    }
                    rowData={Array []}
                    rowSelectedActionCreators={
                      Array [
                        [Function],
                      ]
                    }
                  >
                    <DataGrid
                      columnDefs={
                        Array [
                          Object {
                            "field": "uuid",
                            "headerName": "uuid",
                            "hide": true,
                          },
                          Object {
                            "autoHeight": true,
                            "cellRenderer": [Function],
                            "headerName": "Id",
                          },
                          Object {
                            "field": "name.givenName",
                            "headerName": "Given Name",
                          },
                          Object {
                            "field": "name.familyName",
                            "headerName": "Family Name",
                          },
                          Object {
                            "field": "gender",
                            "headerName": "Gender",
                          },
                          Object {
                            "field": "age",
                            "headerName": "Age",
                          },
                          Object {
                            "headerName": "Checked-in Time",
                            "valueGetter": [Function],
                          },
                        ]
                      }
                      dispatch={[Function]}
                      rowData={Array []}
                      rowSelectedActionCreators={
                        Array [
                          [Function],
                        ]
                      }
                    >
                      <div
                        id="omrsGrid"
                        style={
                          Object {
                            "height": "600px",
                          }
                        }
                      >
                        <label>
                          Search:
                          <input
                            onChange={[Function]}
                            type="text"
                          />
                        </label>
                        <AgGridReact
                          animateRows={true}
                          columnDefs={
                            Array [
                              Object {
                                "field": "uuid",
                                "headerName": "uuid",
                                "hide": true,
                              },
                              Object {
                                "autoHeight": true,
                                "cellRenderer": [Function],
                                "headerName": "Id",
                              },
                              Object {
                                "field": "name.givenName",
                                "headerName": "Given Name",
                              },
                              Object {
                                "field": "name.familyName",
                                "headerName": "Family Name",
                              },
                              Object {
                                "field": "gender",
                                "headerName": "Gender",
                              },
                              Object {
                                "field": "age",
                                "headerName": "Age",
                              },
                              Object {
                                "headerName": "Checked-in Time",
                                "valueGetter": [Function],
                              },
                            ]
                          }
                          enableFiltering={true}
                          enableSorting={true}
                          id="omrsGrid"
                          onFilterChanged={[Function]}
                          onGridReady={[Function]}
                          onRowDataChanged={[Function]}
                          onSelectionChanged={[Function]}
                          rowClassRules="rowClassRules"
                          rowData={Array []}
                          rowSelection="single"
                        >
                          <div
                            style={
                              Object {
                                "height": "100%",
                              }
                            }
                          />
                        </AgGridReact>
                      </div>
                    </DataGrid>
                  </Connect(DataGrid)>
                </div>
              </List>
            </div>
          </ScreeningQueue>
        </Connect(ScreeningQueue)>
      </div>
    </NutritionQueue>
  </Connect(NutritionQueue)>
</Provider>
`;
